plugin/build.gradle.kts

```diff
@@ -83,14 +85,26 @@ sourceSets {
     }
 }
 
+kotlin {
+    @OptIn(ExperimentalBuildToolsApi::class, ExperimentalKotlinGradlePluginApi::class)
+    compilerVersion = libs.versions.kotlin.`for`.gradle.plugin.get()
+}
+
 tasks.compileKotlin {
     compilerOptions {
         optIn.addAll(
                 "kotlinx.benchmark.gradle.internal.KotlinxBenchmarkPluginInternalApi",
                 "kotlin.RequiresOptIn",
         )
-        @Suppress("DEPRECATION")
-        apiVersion = KotlinVersion.KOTLIN_1_4 // the version of Kotlin embedded in Gradle
+        /**
+         * Those versions are configured according to https://docs.gradle.org/current/userguide/compatibility.html
+         * and the Kotlin compiler compatibility policy stating that Kotlin 1.4 is compatible with 1.5 binaries
```

Comment:
Is this information public?

Reply:
Yes. 
https://kotlinlang.org/docs/kotlin-evolution-principles.html#evolving-the-binary-format

> Preferably (but we can't guarantee it), the binary format is mostly forwards compatible with the next language release, but not later ones (in the cases when new features are not used, for example, 1.9 can understand most binaries from 2.0, but not 2.1).

Reply:
Thanks for the clarification!